packages/babel-parser/src/parser/error-message.js

```diff
@@ -1,219 +1,636 @@
-  TupleExpressionHashIncorrectStartSyntaxType:
-    "Tuple expressions starting with '#[' are only allowed when the 'syntaxType' option of the 'recordAndTuple' plugin is set to 'hash'",
-  UnexpectedArgumentPlaceholder: "Unexpected argument placeholder",
-  UnexpectedAwaitAfterPipelineBody:
-    'Unexpected "await" after pipeline body; await must have parentheses in minimal proposal',
-  UnexpectedDigitAfterHash: "Unexpected digit after hash token",
-  UnexpectedImportExport:
-    "'import' and 'export' may only appear at the top level",
-  UnexpectedKeyword: "Unexpected keyword '%0'",
-  UnexpectedLeadingDecorator:
-    "Leading decorators must be attached to a class declaration",
-  UnexpectedLexicalDeclaration:
-    "Lexical declaration cannot appear in a single-statement context",
-  UnexpectedNewTarget: "new.target can only be used in functions",
-  UnexpectedNumericSeparator:
-    "A numeric separator is only allowed between two digits",
-  UnexpectedPrivateField:
-    "Private names can only be used as the name of a class element (i.e. class C { #p = 42; #m() {} } )\n or a property of member expression (i.e. this.#p).",
-  UnexpectedReservedWord: "Unexpected reserved word '%0'",
-  UnexpectedSuper: "super is only allowed in object methods and classes",
-  UnexpectedToken: "Unexpected token '%0'",
-  UnexpectedTokenUnaryExponentiation:
-    "Illegal expression. Wrap left hand side or entire exponentiation in parentheses.",
-  UnsupportedBind: "Binding should be performed on object property.",
-  UnsupportedDecoratorExport:
-    "A decorated export must export a class declaration",
-  UnsupportedDefaultExport:
-    "Only expressions, functions or classes are allowed as the `default` export.",
-  UnsupportedImport: "import can only be used in import() or import.meta",
-  UnsupportedMetaProperty: "The only valid meta property for %0 is %0.%1",
-  UnsupportedParameterDecorator:
-    "Decorators cannot be used to decorate parameters",
-  UnsupportedPropertyDecorator:
-    "Decorators cannot be used to decorate object literal properties",
-  UnsupportedSuper:
-    "super can only be used with function calls (i.e. super()) or in property accesses (i.e. super.prop or super[prop])",
-  UnterminatedComment: "Unterminated comment",
-  UnterminatedRegExp: "Unterminated regular expression",
-  UnterminatedString: "Unterminated string constant",
-  UnterminatedTemplate: "Unterminated template",
-  VarRedeclaration: "Identifier '%0' has already been declared",
-  YieldBindingIdentifier:
-    "Can not use 'yield' as identifier inside a generator",
-  YieldInParameter: "Yield expression is not allowed in formal parameters",
-  ZeroDigitNumericSeparator:
-    "Numeric separator can not be used after leading 0",
+export const ErrorMessages: ErrorTemplates = Object.freeze({
+  AccessorIsGenerator: {
+    code: "AccessorIsGenerator",
+    template: "A %0ter cannot be a generator",
+  },
```

Comment:
It's just a stylistic preference, but I don't really like that we have to repeat so much boilerplate (duplicating every code, and the `code`/`template` keys for every message).

Maybe we could do something like this?

```js
export const ErrorMessages = makeErrorTemplates({
  AccessorIsGenerator: "A %0ter cannot be a generator",
  ...
});

export function makeErrorTemplates(messages) {
  Object.keys(messages).forEach(code => {
    messages[code] = { code, template: messages[code] };
  });
}
```

it's a one-time cost only paid when importing `@babel/parser`.